docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
 title: TurboModules
 ---
 
-This section contains a high-level introduction to TurboModules. It provides enough context to understand when a TurboModule is needed and how it roughly works.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+If you've worked with React Native, you may be familiar with the concept of Native Modules, which allow JavaScript and platform-native code to communicate over the React Native "bridge", which handles cross-platform serialization via JSON.
 
-## How to create a Turbomodule
+TurboModules are the next iteration on Native Modules that provide a few extra [benefits](./why):
 
-This section is a step-by-step guide to create a TurboModule from scratch. The list of subsections is roughly:
+- Strongly typed interfaces that are consistent across platforms
+- The ability to create shared C++ code for use across platforms
```

Comment:
@cortinico can you elaborate for me more what you want to convey by including "C++-only" here? Is there a misunderstanding you're trying to avoid? To my eye, I wouldn't assume the original wording means that I can share code in other languages as well.

(Side note: it would have to be written "C++-only", which looks a bit odd with the "++-", so I would try to reword this once I have a better understanding of intent.

Reply:
I think we are saying two different things:
* In the original formulation, we wanted to convey that we can share code between iOS and Android using C++ code.
* In the suggestion by Nico, we are suggesting that we can create TM that can be used on both platforms and that contains only C++ code (this can be used for TM that do not need to access specific platform's features).

Both statements are true. I would specify both or look for a formulation that implies both meanings. What do you think?

Reply:
Tried to rephrase this to the best of my ability to explain both those ideas (thanks for laying them out so clearly @cipolleschi). Let me know what you think!